### PR TITLE
Cleanup v0.3.0

### PR DESCRIPTION
**Pull request by koaning**
Makes the API scikit-learn compatible again.

### CODE REVIEW

**Review by Proteusiq — APPROVED** [review]
Looks good to me.

File: scripts/demo-annoy.py [hunk 1]
```diff
@@ -4,42 +4,29 @@
...
+from simsity.indexer import AnnoyIndexer
+
+df = pd.read_csv("tests/data/clinc-data.csv").head(100)
+
 # The encoder defines how we encode the data going in.
-encoder = make_pipeline(
-    ColumnLister(column="text"), CountVectorizer(), TruncatedSVD(n_components=10)
-)
+encoder = make_pipeline(CountVectorizer(), TruncatedSVD())
+encoder.fit(df["text"].tolist())
```
Comment (Proteusiq) [inline scripts/demo-annoy.py:20]: Is there a reason for `.to_list()`? Should not `df["text"]` be enough?

Comment (koaning) [reply scripts/demo-annoy.py:20]: Not really. This PR is still a work in progress.

**Comment by koaning** [comment]
While I appreciate folks thinking along, this is very much a "Work In Progress" pr. The docs aren't updated at all yet and the unit tests aren't passing either. The rewrite is also something that isn't set in stone yet and I prefer to iterate on the design personally over the next few months.

**Pull request merged**